test/test_cli.py

```diff
@@ -1,19 +1,40 @@
 """Tests the Kili CLI"""
 
 import os
-from kili.cli import import_assets
+from kili.cli import import_assets, list_project
 from click.testing import CliRunner
 
 from .utils import debug_subprocess_pytest
 
 
-def mocked__projects(project_id, **_):
+def mocked__projects(project_id=None, **_):
     if project_id == 'text_project':
         return [{'id': 'text_project', 'inputType': 'TEXT'}]
     if project_id == 'image_project':
-        return [{'id': 'text_project', 'inputType': 'IMAGE'}]
+        return [{'id': 'image_project', 'inputType': 'IMAGE'}]
     if project_id == 'frame_project':
-        return [{'id': 'text_project', 'inputType': 'FRAME'}]
+        return [{'id': 'frame_project', 'inputType': 'FRAME'}]
+    if project_id == None:
+        return [{'id': 'text_project', 'title': 'text_project', 'description': ' a project with text',
+                 'numberOfAssets': 10, 'numberOfRemainingAssets': 10},
+                {'id': 'image_project', 'title': 'image_project', 'description': ' a project with image',
+                 'numberOfAssets': 0, 'numberOfRemainingAssets': 0},
+                {'id': 'frame_project', 'title': 'frame_project', 'description': ' a project with fram',
+                 'numberOfAssets': 10, 'numberOfRemainingAssets': 0}]
+
+
+def test_list(mocker):
+    mocker.patch("kili.client.Kili.__init__", return_value=None)
+    mocker.patch("kili.client.Kili.projects",
+                 side_effect=mocked__projects)
+
+    runner = CliRunner()
+    with runner.isolated_filesystem():
+        result = runner.invoke(list_project)
+        assert result.exit_code == 0
+        assert ((result.output.count("100.0%") == 1) and
+                (result.output.count("0.0%") == 2) and
+                (result.output.count("nan") == 1))
 
```

Comment:
you can use the assert_called_once on the mocked project query. cf create_project test function

Reply:
I have tried a few ways but none worked, on which mock_patch am I suppose to call assert_called_once?

Reply:
mock__projects = mocker.patch("kili.client.Kili.projects",
                 side_effect=mocked__projects)
...
mock__projects.assert_called_once()

I think that you have to be careful not to call it on `mocked__projects` which is just a function defined to replace the return of projects query. Just find a name for the mock, I called it mock__projects above

Reply:
If you do this you have two mock__projects, the function and the mocker.patch. Is it OK ?

Reply:
I have tried and apparently it is not. And it does not work either when I choose another name... Is it really necessary ? I don't see how I could call the function more than once.

Reply:
Yes, I think that it is ok. 
The tests are also to keep things well in the long time and to avoid someone to decrease performances. if someone for example call one day the query a second time...
Things are good as they are, it was only a very small improvement